watcher/local/watcher_test.go

```diff
@@ -141,6 +141,22 @@ func Test_Watcher_Working(t *testing.T) {
 
 			rs.AssertExpectations(t)
 		})
+
+		t.Run("happy/progressed_event", func(t *testing.T) {
```

Comment:
The tests are very comprehensive (which is good) but could you please add a comment on each subtest what it does? One line would be enough.